include/zephyr/kernel/obj_core.h

```diff
@@ -0,0 +1,283 @@
+extern void k_obj_core_init(struct k_obj_core *obj_core,
+			    struct k_obj_type *type);
+
+/**
+ * @brief Initialize the object type for statistics
+ *
+ * @param type Pointer to the object type
+ * @param desc Pointer to the object type's statistics descriptor
+ */
+extern void k_obj_type_stats_init(struct k_obj_type *type,
+				  struct k_obj_stats_desc *desc);
+
+/**
+ * @brief Link the kernel object to the kernel object type list
+ *
+ * A kernel object can be optionally linked into the kernel object type's
+ * list of objects. A kernel object must have been initialized before it
+ * can be linked. Linked kernel objects can be traversed and have information
+ * extracted from them by system tools.
+ *
+ * @param obj_core Pointer to the kernel object
+ */
+extern void k_obj_core_link(struct k_obj_core *obj_core);
+
+/**
+ * @brief Unlink the kernel object from the kernel object type list
+ *
+ * Kernel objects can be unlinked from their respective kernel object type
+ * lists. If on a list, it must be done at the end of the kernel object's life
+ * cycle.
+ *
+ * @param obj_core Pointer to the kernel object
+ */
+extern void k_obj_core_unlink(struct k_obj_core *obj_core);
+
+/**
+ * @brief Register kernel object for gathering statistics
+ *
+ * Before a kernel object can gather statistics, it must be registered to do
+ * so. Registering will also automatically enable the kernel object to gather
+ * its statistics.
+ *
+ * @param obj_core Pointer to kernel object core
+ * @param stats Pointer to raw kernel statistics
+ * @param stats_len Size of raw kernel statistics buffer
+ *
+ * @return 0 on success, -errno on failure
+ */
+extern int k_obj_stats_register(struct k_obj_core *obj_core, void *stats,
+				size_t stats_len);
+
+/**
+ * @brief Deregister kernel object from gathering statistics
+ *
+ * Deregistering a kernel object core from gathering statistics prevents it
+ * from gathering any more statistics. It is expected to be invoked at the end
+ * of a kernel object's life cycle.
+ *
+ * @param obj_core Pointer to kernel object core
+ *
+ * @return 0 on success, -errno on failure
+ */
+extern int k_obj_stats_deregister(struct k_obj_core *obj_core);
+
+/**
+ * @brief Retrieve the raw statistics associated with the kernel object
+ *
+ * This function copies the raw statistics associated with the kernel object
+ * core specified by @a obj_core into the buffer @a stats. Note that the size
+ * of the buffer (@a stats_len) must match the size specified by the kernel
+ * object type's statistics descriptor.
+ *
+ * @param obj_core Pointer to kernel object core
+ *
+ * @return 0 on success, -errno on failure
+ */
+extern int k_obj_stats_raw(struct k_obj_core *obj_core, void *stats,
+			   size_t stats_len);
+
+/**
+ * @brief Retrieve the statistics associated with the kernel object
+ *
+ * This function copies the statistics (not raw) associated with the kernel
+ * object core specified by @a obj_core into the buffer @a stats. Note that the
```

Comment:
I'm confused.  What's the difference between a "raw" statistics buffer and a "not raw" statistics buffer?

Reply:
I probably should have used "cooked" instead of "not raw". Raw = raw data. Cooked = operations may have been done to the data to make it more palatable for human consumption (calculating averages for example).